### PR TITLE
Testing for Fedora 24

### PR DESCRIPTION
**Pull request by stefwalter**
Remove fedora-22 images and add fedora-24
    
Fedora 24 has branched and we should be treating Fedora 23 is rather stable now ... starting to do more testing against Fedora 24
    
The Atomic and Testing images still track Fedora 23 for now

### CODE REVIEW

**Comment by stefwalter** [comment]
Docker and Kubernetes don't install yet.

File: test/guest/fedora-23.bootstrap [hunk 1]
```diff
@@ -18,4 +18,4 @@
 # 02110-1301 USA.
 
 BASE=$(dirname $0)
-$BASE/fedora.build "$1" fedora-23 x86_64
+$BASE/fedora.build "$1" fedora-24 x86_64
```
Comment (mvollmer) [inline test/guest/fedora-23.bootstrap]: This looks like a mistake.

**Comment by stefwalter** [comment]
Docker now installs.

**Comment by stefwalter** [comment]
@mvollmer Mock fails on fedora-24. Do you have time to take a look?

**Comment by stefwalter** [comment]
```
ERROR: Command failed. See logs for output. # bash --login -c /usr/bin/rpmbuild -bb --target x86_64 --nodeps /builddir/build/SPECS/cockpit.spec WARNING: unable to delete selinux filesystems (/tmp/mock-selinux-plugin.8wy4c7wf): [Errno 1] Operation not permitted: '/tmp/mock-selinux-plugin.8wy4c7wf' Traceback (most recent call last): File "./vm-install", line 97, in <module> build_and_maybe_install(build_image, do_install=True, skip=skip) File "./vm-install", line 71, in build_and_maybe_install run_install_script(machine, True, do_install, skip, os.path.basename(source)) File "./vm-install", line 58, in run_install_script " '%s'" % arg if arg else "")) File "/data/src/cockpit/test/testvm.py", line 360, in execute raise subprocess.CalledProcessError(proc.returncode, command, output=output) subprocess.CalledProcessError: Command 'cd /var/tmp; ./fedora-24.install --build --install --skip 'cockpit-ostree' 'cockpit-wip.tar.gz'' returned non-zero exit status 1 Traceback (most recent call last): File "./verify/run-tests", line 39, in install subprocess.check_call(cmd) File "/usr/lib64/python2.7/subprocess.py", line 540, in check_call raise CalledProcessError(retcode, cmd) CalledProcessError: Command '['./testsuite-prepare']' returned non-zero exit status 1 Install failed
```

File: test/guest/fedora-24.bootstrap [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301 USA.
+
+BASE=$(dirname $0)
+$BASE/fedora.install "$1" x86_64 "http://dl.fedoraproject.org/pub/fedora/linux/development/24/Server/x86_64/os"
```
Comment (dperpeet) [inline test/guest/fedora-24.bootstrap]: Is there a reason you don't use fedora.build here?

Comment (stefwalter) [inline test/guest/fedora-24.bootstrap]: There are no virt-builder images until it's released.

**Comment by mvollmer** [comment]
> @mvollmer Mock fails on fedora-24. Do you have time to take a look?

I start now.

**Comment by mvollmer** [comment]
- npm fails inside mock somehow, but ust skipping it works
- GCC has new warnings and one of them bites
- one unit test about ssk keys produces different output
- many packages are not signed

**Comment by mvollmer** [comment]
> - npm fails inside mock somehow, but just skipping it works

This must have been some local problem with my source tree... Removing node_modules and recreating it fixed this.  However, I think we should remove "npm install" from autogen.sh, since node_modules is now in git.

**Comment by mvollmer** [comment]
Continued in #3936

**Pull request closed without merging**